bundles/ch.itmed.lmz.risch.laborder/src/ch/itmed/lmz/risch/laborder/net/HttpsConnectionHandler.java

```diff
@@ -0,0 +1,167 @@
+ *
+ * Contributors:
+ *     IT-Med AG <info@it-med-ag.ch> - initial implementation
+ ******************************************************************************/
+
+package ch.itmed.lmz.risch.laborder.net;
+
+import java.awt.Desktop;
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.NoRouteToHostException;
+import java.net.URI;
+import java.net.URL;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.security.UnrecoverableKeyException;
+import java.util.Base64;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+import ch.itmed.lmz.risch.laborder.json.ApiRequest;
+import ch.itmed.lmz.risch.laborder.json.ApiResponse;
+import ch.itmed.lmz.risch.laborder.preferences.PreferenceConstants;
+import ch.itmed.lmz.risch.laborder.preferences.SettingsProvider;
+import ch.itmed.lmz.risch.laborder.ui.MessageBoxUtil;
+
+public final class HttpsConnectionHandler {
+
+	private static Logger logger = LoggerFactory.getLogger(HttpsConnectionHandler.class);
+
+	public static void connect(final String formId) {
+
+		KeyStore clientStore = null;
+		try {
+			clientStore = KeyStore.getInstance("PKCS12");
+			clientStore.load(
+					new FileInputStream(SettingsProvider.getSettings().get(PreferenceConstants.CLIENT_CERTIFICATE, "")),
+					SettingsProvider.getSettings().get(PreferenceConstants.CERTIFICATE_PASSWORD, "").toCharArray());
+		} catch (FileNotFoundException e) {
+			MessageBoxUtil.showErrorDialog("Zertifikat nicht gefunden",
+					"Das angegebene Zertifikat konnte nicht geöffnet");
+			logger.error("Certificate not found", e);
+		} catch (IOException e) {
+			if (e.getMessage().equals("keystore password was incorrect")) {
+				MessageBoxUtil.showErrorDialog("Zertifikat nicht geladen",
+						"Das Zertifikat-Passwort wurde nicht akzeptiert");
+			}
+			logger.error("Configuration error", e);
+		} catch (Exception e) {
+			logger.error("General error", e);
+		}
+
+		try {
+			/*
+			 * KeyManagers are responsible for managing the key material which is used to
+			 * authenticate the local SSLSocket to its peer.
+			 */
+			KeyManagerFactory keyManagerFactory = KeyManagerFactory
+					.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+			keyManagerFactory.init(clientStore,
+					SettingsProvider.getSettings().get(PreferenceConstants.CERTIFICATE_PASSWORD, "").toCharArray());
+
+			KeyManager[] keyManagers = keyManagerFactory.getKeyManagers();
+
+			// Load the Windows Trusted Root Store
+			KeyStore winKeystore = KeyStore.getInstance("Windows-ROOT");
```

Comment:
@col-panic Es wird lediglich die Windows Root CA ausgelesen. Seitens IT-Med nutzt niemand eine andere Plattform, daher wurde es auch entsprechend implementiert.